### PR TITLE
Prevent illegal server info from GUI

### PR DESCRIPTION
**Pull request by pljones**

### CODE REVIEW

File: src/settings.h [hunk 1]
```diff
@@ -181,7 +181,7 @@ class CClientSettings : public CSettings
 class CServerSettings : public CSettings
 {
 public:
-    CServerSettings ( CServer* pNSerP, const QString& sNFiName ) : CSettings(), pServer ( pNSerP )
+    CServerSettings ( CServer* pNSerP, const QString& sNFiName ) : CSettings(), pServer ( pNSerP ), reSvrInfoIllegal ( QRegExp ( "[!\n;]" ) )
```
Comment (softins) [inline src/settings.h:184]: See comment above for `serverdlg.cpp`. I realise this particular RE is only used for `QString::remove()`, which would imply you do actually intend to remove `!` characters too. (?)

**Review by softins — CHANGES_REQUESTED** [review]
Comments made from reading, without trying to compile and execute.

File: src/serverdlg.cpp [hunk 2]
```diff
@@ -30,7 +30,8 @@ CServerDlg::CServerDlg ( CServer* pNServP, CServerSettings* pNSetP, const bool b
     pServer ( pNServP ),
     pSettings ( pNSetP ),
     BitmapSystemTrayInactive ( QString::fromUtf8 ( ":/png/LEDs/res/CLEDGreyArrow.png" ) ),
-    BitmapSystemTrayActive ( QString::fromUtf8 ( ":/png/LEDs/res/CLEDGreenArrow.png" ) )
+    BitmapSystemTrayActive ( QString::fromUtf8 ( ":/png/LEDs/res/CLEDGreenArrow.png" ) ),
+    reSvrInfoIllegal ( QRegExp ( "[!;\n]*" ) )
```
[the remark this answers is not in the document]
Comment (ann0see) [reply src/serverdlg.cpp:34]: ! does't seem to be an invalid character (at least I doubt it's a problem). Waiting on @pljones

**Review by ann0see — COMMENTED** [review]
Was this PR raised to remain consistent with the CLI (which doesn't support \n and ;) or is it because these characters are problematic in any other way?

File: src/serverdlg.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (pljones) [reply src/serverdlg.cpp:34]: Should be "^" at tge start.

**Comment by pljones** [comment]
Hopefully it should be a lot better today.

(I tested this one, too...)

**Comment by pljones** [comment]
Going with a different approach, so this solution isn't needed.

**Pull request closed without merging**